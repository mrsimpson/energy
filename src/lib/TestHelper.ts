import { createRouter, createWebHistory, type RouteRecordRaw } from 'vue-router';
import { NModal } from "naive-ui";
import type { Component } from 'vue';
import { createTestingPinia } from "@pinia/testing";
import { vi } from 'vitest';

type Modal = { componentVM: { containerRef: HTMLElement } }

export default function TestHelper() {
  let modalContainerCreated = false
  const plugins = [] as any[]

  return {
    setupStore() {
      const store = createTestingPinia({ createSpy: vi.fn });
      plugins.push(store)
    },

    async setupRouter(routes: Array<RouteRecordRaw> = []) {
      const router = createRouter({
        history: createWebHistory(),
        routes,
      })
      router.push('/')
      await router.isReady()
      plugins.push(router)
      return router
    },
        
    setupModal() {
      const el = document.createElement("div");
      el.id = "n-modal-container";
      document.body.appendChild(el);
      modalContainerCreated = true
    },

    teardown() {
      if (modalContainerCreated) {
        document.body.outerHTML = "";
      }
    },

    getModalElement(wrapper: { getComponent(component: Component): unknown }) {
      if (!modalContainerCreated) {
        return null
      }
      const modal = wrapper.getComponent(NModal) as Modal
      return modal.componentVM.containerRef.firstChild;
    },

    mountOptions() {
      return {
        global: { plugins },
      }
    }
  }
}
