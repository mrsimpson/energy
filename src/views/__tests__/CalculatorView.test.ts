import { describe, expect, it } from "vitest";
import { shallowMount } from "@vue/test-utils";
import TestHelper from "@/lib/TestHelper";
import component from "../CalculatorView.vue";

describe("CalculatorView", () => {
  it("renders as expected", () => {
    const testHelper = TestHelper();
    testHelper.setupStore();
    const wrapper = shallowMount(component, testHelper.mountOptions());
    expect(wrapper.element).toMatchSnapshot();
  });
});
