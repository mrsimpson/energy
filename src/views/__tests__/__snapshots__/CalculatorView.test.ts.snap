// Vitest Snapshot v1

exports[`CalculatorView > renders as expected 1`] = `
<main>
  <the-gas-calculator-stub />
</main>
`;
